landlab/components/lateral_erosion/lateral_erosion.py

```diff
@@ -0,0 +1,683 @@
+            # to 0.
+            self.qsinlet = qsinlet
+            self.qs_in[self.inlet_node] = self.qsinlet
+
+        # handling Kv for floats (inwhich case it populates an array N_nodes long) or
+        # for arrays of Kv. Checks that length of Kv array is good.
+        self.Kv = np.ones(self.grid.number_of_nodes, dtype=float) * Kv
+
+    def run_one_step(self, *args, **kwds):
+        return self._run_one_step(*args, **kwds)
+
+    def run_one_step_basic(
+        self, grid, dt=None, Klr=None, inlet_area_ts=None, qsinlet_ts=None, **kwds
+    ):
+        if Klr is None:  # Added10/9 to allow changing rainrate (indirectly this way.)
+            Klr = self.Klr
+
+        UC = self._UC
+        TB = self._TB
+        inlet_on = self.inlet_on  # this is a true/false flag
+        Kv = self.Kv
+        qs_in = self.qs_in
+        dzdt = self.dzdt
+        alph = self.alph
+        self.dt = dt
+        vol_lat = self.grid.at_node["volume__lateral_erosion"]
+        kw = 10.0
+        F = 0.02
+        # May 2, runoff calculated below (in m/s) is important for calculating
+        # discharge and water depth correctly. renamed runoffms to prevent
+        # confusion with other uses of runoff
+        runoffms = (Klr * F / kw) ** 2
+        # Kl is calculated from ratio of lateral to vertical K parameters
+        Kl = Kv * Klr
+        z = grid.at_node["topographic__elevation"]
+        # clear qsin for next loop
+        qs_in = grid.add_zeros("node", "qs_in", noclobber=False)
+        qs = grid.add_zeros("node", "qs", noclobber=False)
+        lat_nodes = np.zeros(grid.number_of_nodes, dtype=int)
+        dzlat = np.zeros(grid.number_of_nodes)
+        dzver = np.zeros(grid.number_of_nodes)
+        vol_lat_dt = np.zeros(grid.number_of_nodes)
+        if inlet_on is True:
+            inlet_node = self.inlet_node
+            # if a value is passed with qsinlet_ts, qsinlet has changed with this timestep,
+            # so reset qsinlet to qsinlet_ts
+            if qsinlet_ts is not None:
+                qsinlet = qsinlet_ts
+                qs_in[inlet_node] = qsinlet
+            # if nothing is passed with qsinlet_ts, qsinlet remains the same from
+            # initialized parameters
+            else:  # qsinlet_ts==None:
+                qsinlet = self.qsinlet
+                qs_in[inlet_node] = qsinlet
+            if inlet_area_ts is not None:
+                inlet_area = inlet_area_ts
+                runoffinlet = np.ones(grid.number_of_nodes) * grid.dx ** 2
+                # Change the runoff at the inlet node to node area + inlet node
+                runoffinlet[inlet_node] += inlet_area
+                _ = grid.add_field(
+                    "node", "water__unit_flux_in", runoffinlet, noclobber=False
+                )
+                # if inlet area has changed with time (so we have a new inlet area here)
+                fa = FlowAccumulator(
```

Comment:
Instead of creating a FlowAccumulator *within* your component, better practice would be to have your model (i.e., the driver program) create a FlowAccumulator, and then the lateral erosion component just uses whatever it needs as input fields (and they should be listed in the header). That should also speed it up a lot, since it takes time to instantiate a new FlowAccumulator with each time step.

Reply:
I removed two instances where a FlowAccumulator was created, but kept the creation of a FlowAccumulator with adaptive time stepping for now. I want to test removing this and maybe we can talk about it.

Reply:
@alangston, I put some and recommendations in the code review on this subject.